vmdb/app/services/orchestration_template_dialog_service.rb

```diff
@@ -0,0 +1,151 @@
+class OrchestrationTemplateDialogService
+  def create_dialog(dialog_label, template)
+    ActiveRecord::Base.transaction do
+      dialog = Dialog.create(:label => dialog_label, :buttons => "submit,cancel")
+      tab = DialogTab.create(:display => "edit", :label => "Basic Information", :dialog => dialog, :position => 0)
+      add_stack_group(tab, 0)
+
+      template.parameter_groups.each_with_index do |parameter_group, index|
+        add_parameter_group(parameter_group, tab, index + 1)
+      end
+      dialog
+    end
+  end
+
+  private
+
+  def add_stack_group(tab, position)
+    dialog_group = DialogGroup.create(
+      :display    => "edit",
+      :label      => "Options",
+      :dialog_tab => tab,
+      :position   => position
+    )
+    add_provider_field(dialog_group, 0)
+    add_stack_name_field(dialog_group, 1)
+    add_on_failure_field(dialog_group, 2)
+    add_timeout_field(dialog_group, 3)
+    dialog_group
+  end
+
+  def add_parameter_group(parameter_group, tab, position)
+    dialog_group = DialogGroup.create(
+      :display    => "edit",
+      :label      => parameter_group.label,
+      :dialog_tab => tab,
+      :position   => position
+    )
+    parameter_group.parameters.each_with_index { |param, index| add_parameter_field(param, dialog_group, index) }
+    dialog_group
+  end
+
+  def add_provider_field(group, position)
+    DialogFieldDynamicList.create(
+      :name         => "stack_provider",
+      :description  => "Select a provider that will generate this stack",
+      :display      => "edit",
+      :required     => true,
+      :options      => {:load_values_on_init => false, :show_refresh_button => false},
+      :label        => "Provider",
+      :position     => position,
+      :dialog_group => group
+    )
+  end
+
+  def add_stack_name_field(group, position)
+    DialogFieldTextBox.create(
+      :name           => "stack_name",
+      :description    => "Name of the stack",
+      :data_type      => "string",
+      :display        => "edit",
+      :required       => true,
+      :options        => {:protected => false},
+      :validator_type => 'regex',
+      :validator_rule => '^[A-Za-z][A-Za-z0-9|-]*$',
+      :label          => "Stack Name",
+      :position       => position,
+      :dialog_group   => group
+    )
```

Comment:
Subclasses models need a little more work as the type property needs to be passed.  We will want to include the NewWithTypeStiMixin into the dialog_field.rb model.

Then this becomes:
```ruby
group.dialog_fields.build(
  :type => "DialogFieldTextBox"
  :name           => "stack_name",
  :description    => "Name of the stack",
  :data_type      => "string",
  :display        => "edit",
  :required       => true,
  :options        => {:protected => false},
  :validator_type => 'regex',
  :validator_rule => '^[A-Za-z][A-Za-z0-9|-]*$',
  :label          => "Stack Name",
  :position       => position,
)
```

Same basic change for the other dialog_field methods.